### PR TITLE
Show alert when downloading nb with env embedded

### PR DESCRIPTION
**Pull request by malev**
If the notebook has an environment attached, show this message:
```
➜  tmp  anaconda notebook download malev/xkcd_density
Using binstar api site https://api.beta.anaconda.org
malev/xkcd_density has been downloaded as xkcd_density.ipynb.
xkcd_density.ipynb has an environment embedded.
Run:
    conda env create xkcd_density.ipynb
To install the environment in your system
```
@srossross @ijstokes @goanpeca

### CODE REVIEW

File: binstar_client/utils/notebook/__init__.py [hunk 1]
```diff
@@ -34,3 +39,11 @@ def notebook_url(upload_info):
     else:
         url = "{}://{}/notebooks{}".format(parsed.scheme, parsed.netloc, parsed.path)
     return url
+
+
+def has_environment(nb_file):
+    try:
+        nb = nbformat.reader.reads(open(nb_file).read())
+        return 'environment' in nb['metadata']
+    except (AttributeError, TypeError):
```
Comment (goanpeca) [inline binstar_client/utils/notebook/__init__.py]: Its longer... but isnt it "better"?

```python
with open(nb_file) as f:
    data = f.read()
nb = nbformat.reader.reads(data)
```

Comment (srossross) [inline binstar_client/utils/notebook/__init__.py]: :+1: yes, this is the way we should open files in all cases.

Comment (srossross) [inline binstar_client/utils/notebook/__init__.py]: To be clear @goanpeca yes you are correct.

File: binstar_client/utils/notebook/__init__.py [hunk 2]
```diff
@@ -34,3 +39,11 @@ def notebook_url(upload_info):
     else:
         url = "{}://{}/notebooks{}".format(parsed.scheme, parsed.netloc, parsed.path)
     return url
+
+
+def has_environment(nb_file):
+    try:
```
Comment (goanpeca) [inline binstar_client/utils/notebook/__init__.py]: if `nbformat` is defined as `None` if the import fails... why not use that in an `if`, 

```python
if nbformat:
    ...
```

Comment (malev) [inline binstar_client/utils/notebook/__init__.py]: yes but I like the idea of having everything within a `try/catch` block.

File: binstar_client/utils/notebook/__init__.py [hunk 3]
```diff
@@ -34,3 +39,15 @@ def notebook_url(upload_info):
...
+        with open(nb_file) as fb:
+            data = fb.read()
+        nb = nbformat.reader.reads(data)
+        return 'environment' in nb['metadata']
+    except AttributeError:
+        return False
+    except IOError:
+        raise BinstarErro("{} can't be opened".format(nb_file))
```
Comment (goanpeca) [inline binstar_client/utils/notebook/__init__.py]: Is this a typo?

**Comment by goanpeca** [comment]
@malev maybe some tests should be added...

**Comment by goanpeca** [comment]
LGTM :+1:

**Pull request merged**